seatunnel-connectors-v2/connector-dingtalk/src/main/java/org/apache/seatunnel/connectors/seatunnel/source/DingTalkSourceReader.java

```diff
@@ -0,0 +1,89 @@
+import org.apache.seatunnel.api.table.type.SeaTunnelRow;
+import org.apache.seatunnel.common.utils.JsonUtils;
+import org.apache.seatunnel.connectors.seatunnel.common.DingTalkConstant;
+import org.apache.seatunnel.connectors.seatunnel.common.DingTalkParameter;
+import org.apache.seatunnel.connectors.seatunnel.common.source.AbstractSingleSplitReader;
+import org.apache.seatunnel.connectors.seatunnel.common.source.SingleSplitReaderContext;
+
+import com.dingtalk.api.DefaultDingTalkClient;
+import com.dingtalk.api.DingTalkClient;
+import com.dingtalk.api.request.OapiV2DepartmentListsubRequest;
+import com.dingtalk.api.response.OapiV2DepartmentListsubResponse;
+import com.fasterxml.jackson.databind.JsonNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+public class DingTalkSourceReader extends AbstractSingleSplitReader<SeaTunnelRow> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DingTalkSourceReader.class);
+    protected final SingleSplitReaderContext context;
+    protected final DingTalkParameter dtParameter;
+    protected DingTalkClient dtClient;
+    protected OapiV2DepartmentListsubRequest dtRequest;
+    protected final DeserializationSchema<SeaTunnelRow> deserializationSchema;
+
+    public DingTalkSourceReader(DingTalkParameter dtParameter, SingleSplitReaderContext context, DeserializationSchema<SeaTunnelRow> deserializationSchema) {
+        this.context = context;
+        this.dtParameter = dtParameter;
+        this.deserializationSchema = deserializationSchema;
+    }
+
+    @Override
+    public void open() {
+        dtClient = new DefaultDingTalkClient(dtParameter.getApiClient());
+        dtRequest = new OapiV2DepartmentListsubRequest();
+        LOGGER.info("Ding Talk Access Token is :" + dtParameter.getAccessToken());
+    }
+
+    @Override
+    public void close() throws IOException {
+    }
+
+    @Override
+    public void pollNext(Collector<SeaTunnelRow> output) {
+        try {
+            OapiV2DepartmentListsubResponse response = dtClient.execute(dtRequest, dtParameter.getAccessToken());
+            if (DingTalkConstant.STATUS_OK.equals(response.getErrmsg())) {
+                String tmpContent = response.getBody();
+                JsonNode bodyJson = JsonUtils.stringToJsonNode(tmpContent);
+                JsonNode resJson = bodyJson.get(DingTalkConstant.BODY_RESULT);
+                if (resJson.isArray()) {
+                    for (JsonNode tmpJson : resJson) {
+                        output.collect(new SeaTunnelRow(new Object[]{tmpJson.toString()}));
+                    }
+                }
+            }
+            LOGGER.error("Ding Talk client execute exception, response status code:[{}], content:[{}]", response.getErrorCode(), response.getBody());
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage(), e);
+        } finally {
+            if (Boundedness.BOUNDED.equals(context.getBoundedness())) {
```

Comment:
It can be seen from this code that the connector supports both batch and stream mode. But from the document I see the `stream` is not been selected. Please check this.